### PR TITLE
Run tests using Karma

### PR DESCRIPTION
**Pull request by SimenB**
Hey,

Using karma to run the test makes it much easier to extract coverage, run in multiple browsers, and run from the IDE (if using WebStorm/IntelliJ). It also has other features such as watching files, etc.

As mentioned in #861 the tests fail, but you said you were on it, so I haven't looked into it

### CODE REVIEW

File: qunit/karmaQunitSetup.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+QUnit.begin(function () {
```
Comment (SimenB) [inline qunit/karmaQunitSetup.js]: https://github.com/karma-runner/karma-qunit/issues/18

File: karma.conf.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        // frameworks to use
+        // available frameworks: https://npmjs.org/browse/keyword/karma-adapter
+        frameworks: ['qunit'],
+
+
+        // list of files / patterns to load in the browser
+        files: [
+            'node_modules/jquery/dist/jquery.js',
+            //'js/**/*.js',
+            'dist/jquery.inputmask.bundle.js',
```
Comment (SimenB) [inline karma.conf.js]: We could possibly look into always having the UMD wrapper in the source, and not injecting at build time. Then using e.g. https://github.com/Nikku/karma-browserify.
Then we could have the real source tested, and not have to fetch jquery manually from `node_modules`

File: package.json [hunk 3]
```diff
@@ -4,18 +4,14 @@
   "description": "jquery.inputmask is a jquery plugin which create an input mask.",
   "main": "./dist/inputmask/jquery.inputmask.js",
   "files": [
-    "dist/inputmask/jquery.inputmask.js",
-    "dist/inputmask/inputmask.js",
-    "dist/inputmask/inputmask.extensions.js",
     "dist/inputmask/inputmask.date.extensions.js",
```
Comment (SimenB) [inline package.json:7]: npm automatically alphabetizes lists in package.json.

I also included the whole `extra/phone-codes/` directory. The reason that the js-files are listed explicitly is that minified is in the same directory

File: package.json [hunk 4]
```diff
@@ -25,12 +21,13 @@
...
   "keywords": [
-    "jquery",
-    "plugins",
-    "input",
     "form",
+    "input",
     "inputmask",
-    "mask"
+    "jquery",
+    "jquery-plugin",
```
Comment (SimenB) [inline package.json:28]: Alphabetized, and added `jquery-plugin`

**Comment by SimenB** [comment]
@RobinHerbots Updated and rebased this :smile: 
There are some test-failures. Tests are passing in Chrome, but failing with Phantom... Any idea? It's the same is it is currently though, so I think karma is set up correctly

**Comment by RobinHerbots** [comment]
@SimenB , 

Perfect thx !!

**Comment by RobinHerbots** [comment]
@SimenB , 

Karma seems a very nice testing framework.  With the chrome launcher all tests pass.

**Comment by SimenB** [comment]
Yes it works with Chrome. That can't be run by Travis though. A bit annoying... But at least it can be run from the comman line now :smile:

**Comment by RobinHerbots** [comment]
@SimenB , 

Hmm yes, maybe I need to setup travis ci also.  I will have a look why they all fail within phantomjs.

**Pull request merged**